### PR TITLE
#471 FIX

### LINKED ISSUE #530: Java Client, Appium and Win 10 issue
I am trying to run Calculator App (java version) with WinAppDriver 0.5, but I see following issues with Appium.

When I declare my variables as IOSDriver (As mentioned on the description)

private static IOSDriver CalculatorSession = null;
 private static WebElement CalculatorResult = null;
 with following capabilities

capabilities.setCapability("app", "Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
 capabilities.setCapability("platformName", "Windows");
 capabilities.setCapability("deviceName", "WindowsPC");

I keep getting iOS driver and exception
 org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: Could not find path to Xcode by symlinks located in /var/db/xcode_select_link, or /usr/share/xcode-select/xcode_dir_link (WARNING: The server did not provide any stacktrace information)

Now if I change it to
 private static RemoteWebDriver CalculatorSession = null;
 private static RemoteWebElement CalculatorResult = null;

It starts working but now WinAppDriver fails to getText from screen. Following line fails
 final String xPath = "//*[@AutomationId='CalculatorResults']";
 String text = CalculatorSession.findElementByXPath(xPath).getText();

I tried changing xpath to different ways like
 //windows[@name='Display is 7 '];
 //*[@name='Display is 7 '];
 but none worked.
 however when I call CalculatorSession.findElementByName("Seven").click();, this run successfully.

Can someone suggest what's wrong in here? To me it looks like driver issue.

Env:
 Windows 10 Laptop
 Version 1511
 OS Build: 10586.679
 Appium Ver: 1.6.1
 Win App Driver: 0.5

This can be reproduce with Calculator Java version of App from WindowsAppDriver Sample tests.

### PR DESCRIPTION
**Pull request by TikhomirovSergey**
## Change list
- `WindowsDriver` was designed
- `WindowsElement` was designed
- `WindowsKeyCode` was added
- Page object tools were updated:
  - `WindowsFindBy` was added
  - `AppiumFieldDecorator` and supporting tools were actualized

  
Additional change:
- All constructors declared by `AppiumDriver` are public now.
 
## Types of changes

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#471 FIX
#530 FIX

### CODE REVIEW

**Comment by TikhomirovSergey** [comment]
ping @SrinivasanTarget 
@yodurr @jonstoneman you are invited to review this PR too

File: src/main/java/io/appium/java_client/AppiumDriver.java [hunk 1]
```diff
@@ -335,7 +335,7 @@ public void zoom(int x, int y) {
     @Override public String getContext() {
         String contextName =
             String.valueOf(execute(DriverCommand.GET_CURRENT_CONTEXT_HANDLE).getValue());
-        if (contextName.equals("null")) {
+        if ("null".equalsIgnoreCase(String.valueOf(contextName))) {
```
Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: Cant this be done using java Optional class?

Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: ```contextName``` is already a string right then why are we using valueof() again?

File: src/main/java/io/appium/java_client/internal/ElementMap.java [hunk 2]
```diff
@@ -34,7 +36,8 @@
     YOUI_ENGINE(AutomationName.YOUI_ENGINE.toLowerCase(), YouiEngineElement.class),
     IOS_XCUI_TEST(AutomationName.IOS_XCUI_TEST.toLowerCase(), IOSElement.class),
     ANDROID_UI_AUTOMATOR(MobilePlatform.ANDROID.toLowerCase(), AndroidElement.class),
-    IOS_UI_AUTOMATION(MobilePlatform.IOS.toLowerCase(), IOSElement.class);
+    IOS_UI_AUTOMATION(MobilePlatform.IOS.toLowerCase(), IOSElement.class),
+    WINDOwS(MobilePlatform.WINDOWS, WindowsElement.class);
```
Comment (yodurr) [inline src/main/java/io/appium/java_client/internal/ElementMap.java]: lowercase w?

File: src/main/java/io/appium/java_client/pagefactory/DefaultElementByBuilder.java [hunk 3]
```diff
@@ -183,29 +183,29 @@ private static void checkDisallowedAnnotationPairs(Annotation a1, Annotation a2)
             iOSFindAll iOSFindAll = annotatedElement.getAnnotation(iOSFindAll.class);
 
             if (iOSFindByArray != null && iOSFindByArray.length == 1) {
-                return createBy(new Annotation[] {iOSFindByArray[0]}, HowToUseSelectors.USE_ONE);
+                return  createBy(new Annotation[] {iOSFindByArray[0]}, HowToUseSelectors.USE_ONE);
```
Comment (yodurr) [inline src/main/java/io/appium/java_client/pagefactory/DefaultElementByBuilder.java]: probably should undo the extra space

**Comment by yodurr** [comment]
This work is great to see, thank you Sergey!

File: src/main/java/io/appium/java_client/internal/ElementMap.java (hunk 2 above)
Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/internal/ElementMap.java]: @yodurr It is ok. According to [Java conversation rules](http://docs.oracle.com/javase/tutorial/java/javaOO/enum.html) enum elements usully have upper case name. :)
Thank you.

Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/internal/ElementMap.java]: Ok. Maybe it is excessive :)

Comment (yodurr) [inline src/main/java/io/appium/java_client/internal/ElementMap.java]: sorry, I mean why is it "WINDOwS" on line 40?  Shouldn't it be all caps?

Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/internal/ElementMap.java]: Yup :)

Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/internal/ElementMap.java]: Ok. I will improve that :)

**Pull request merged**